src/vos/evtree.c

```diff
@@ -1411,19 +1400,44 @@ evt_root_destroy(struct evt_context *tcx)
 	return evt_root_free(tcx);
 }
 
+static int64_t
+evt_epoch_dist(struct evt_context *tcx, struct evt_node *nd,
+	       const struct evt_rect *rect)
+{
+	struct evt_rect	*mbr = evt_node_mbr_get(tcx, nd);
+
+	if (mbr->rc_epc > rect->rc_epc)
+		return mbr->rc_epc - rect->rc_epc;
+
+	return rect->rc_epc - mbr->rc_epc;
+}
+
 /** Select a node from two for the rectangle \a rect being inserted */
 static struct evt_node *
 evt_select_node(struct evt_context *tcx, const struct evt_rect *rect,
 		struct evt_node *nd1, struct evt_node *nd2)
 {
 	struct evt_weight	wt1;
 	struct evt_weight	wt2;
+	uint64_t		dist1;
+	uint64_t		dist2;
 	int			rc;
 
 	evt_node_weight_diff(tcx, nd1, rect, &wt1);
 	evt_node_weight_diff(tcx, nd2, rect, &wt2);
 
 	rc = evt_weight_cmp(&wt1, &wt2);
+
+	if (rc == 0) {
+		dist1 = evt_epoch_dist(tcx, nd1, rect);
+		dist2 = evt_epoch_dist(tcx, nd2, rect);
+
+		if (dist1 < dist2)
+			return nd1;
+		else
+			return nd2;
+	}
```

Comment:
If there are two full overwrite nodes, node1 have epoch 1, 2, 3, 4, node2 have epoch 6, 7, 8, 9 10, 11, then an epoch 5 overwrite comes, I suppose node1 should be the better choice but we'll chose node2 according to the epoch distance?

Reply:
yes, that is true.   Without a high epoch in the mbr, there is no perfect heuristic